crates/ruff_python_parser/src/error.rs

```diff
@@ -638,6 +638,33 @@ pub enum UnsupportedSyntaxErrorKind {
     ///
     /// [PEP 646]: https://peps.python.org/pep-0646/#change-1-star-expressions-in-indexes
     StarExpressionInIndex,
+
+    /// Represents the use of a [PEP 646] star annotations in a function definition.
+    ///
+    /// ## Examples
+    ///
+    /// Before Python 3.11, star annotations were not allowed in function definitions. This
+    /// restriction was lifted in [PEP 646] to allow type annotations for `typing.TypeVarTuple`,
+    /// also added in Python 3.11:
+    ///
+    /// ```python
+    /// from typing import TypeVarTuple
+    ///
+    /// Ts = TypeVarTuple('Ts')
+    ///
+    /// def foo(*args: *Ts): ...
+    /// ```
+    ///
+    /// Unlike [`UnsupportedSyntaxErrorKind::StarExpressionInIndex`], this does not include any
+    /// other annotation positions:
+    ///
+    /// ```python
+    /// x: *Ts                # Syntax error
+    /// def foo(x: *Ts): ...  # Syntax error
+    /// ```
+    ///
+    /// [PEP 646]: https://peps.python.org/pep-0646/#change-1-star-expressions-in-indexes
```

Comment:
The link here is incorrect. I think it should point to `https://peps.python.org/pep-0646/#change-2-args-as-a-typevartuple`